include/boost/json/detail/number.hpp

```diff
@@ -39,6 +39,7 @@ class number_parser
     enum class state
     {
         init,   init0,  init1,
+        mant_frac_1_digit,      // mantissa: after decimal there must be a digit
```

Comment:
`mantf` would be more in line with the style, and there's no need for the comment here since there is one in the case label (the `1*digit`)